enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DependencyEnricher.java

```diff
@@ -182,6 +178,32 @@ private void removeTemplateObjects(List<HasMetadata> list, List<HasMetadata> obj
 
     }
 
+    public void filterAndAddItemsToBuilder(KubernetesListBuilder builder, List<HasMetadata> items) {
+        HashMap<KindAndName, Integer> aIndexMap = new HashMap<>();
+        int nItems = 0;
+
+        // Populate map with existing items in the builder
+        for(int index = 0; index < builder.getItems().size(); index++, nItems++) {
+            HasMetadata aItem = builder.getItems().get(index);
+            KindAndName aKey = new KindAndName(aItem);
+            aIndexMap.put(aKey, index);
+        }
+
+        for(HasMetadata item : items) {
+            KindAndName aKey = new KindAndName(item);
+
+            if(aIndexMap.containsKey(aKey)) { // Merge the override fragments, and remove duplicate
+                HasMetadata duplicateItem = builder.getItems().get(aIndexMap.get(aKey));
```

Comment:
insted of `builder.getItems().get(aIndexMap.get(aKey));` use `aIndexMap.get(aKey)`